Purchases/RCBackend.m

```diff
@@ -393,9 +413,69 @@ - (void)createAliasForAppUserID:(NSString *)appUserID
                                        @"new_app_user_id": newAppUserID
                                }
                             headers:self.headers
-                  completionHandler:^(NSInteger status, NSDictionary *response, NSError *error) {
+                  completionHandler:^(NSInteger status, NSDictionary *_Nullable response, NSError *_Nullable error) {
                       [self handle:status withResponse:response error:error errorHandler:completion];
                   }];
 }
 
+
+- (void)postOfferForSigning:(NSString *)offerIdentifier
+      withProductIdentifier:(NSString *)productIdentifier
+          subscriptionGroup:(NSString *)subscriptionGroup
+                       data:(NSData *)data
+                  appUserID:(NSString *)appUserID
+                 completion:(RCOfferSigningResponseHandler)completion
+{
+    NSString *fetchToken = [data base64EncodedStringWithOptions:0];
+    [self.httpClient performRequest:@"POST" path:@"/offers"
+                               body:@{
+                                       @"app_user_id": appUserID,
+                                       @"fetch_token": fetchToken,
+                                       @"generate_offers": @[@{
+                                               @"offer_id": offerIdentifier,
+                                               @"product_id": productIdentifier,
+                                               @"subscription_group": subscriptionGroup
+                                       }],
+                               }
+                            headers:self.headers
+                  completionHandler:^(NSInteger statusCode, NSDictionary *_Nullable response, NSError *_Nullable error) {
+                      if (error != nil) {
+                          completion(nil, nil, nil, nil, [RCPurchasesErrorUtils networkErrorWithUnderlyingError:error]);
+                          return;
+                      }
+
+                      NSArray *offers = nil;
+
+                      if (statusCode < 300) {
+                          offers = response[@"offers"];
+                          if (offers == nil || offers.count == 0) {
+                              error = [RCPurchasesErrorUtils unexpectedBackendResponseError];
+                          } else {
+                            NSDictionary *offer = offers[0];
+                            if (offer[@"signature_error"] != [NSNull null]) {
```

Comment:
So this will break if the `signature_error` key is missing from the response, which could be quite confusing. I would do a check like 

`if (offer[@"signature_error"] != nil || offer[@"signature_error"] != [NSNull null])`

Could combine it into a simple macro:
`RC_HAS_KEY(offer, @"signature_error")`

Something like that?